dpctl/_init_helper.py

```diff
@@ -22,6 +22,8 @@
     # with DPC++ libraries to the DLL search path gh-1745
     if "VIRTUAL_ENV" in os.environ:
         venv_dir = os.environ["VIRTUAL_ENV"]
-        expected_dir = os.path.join(venv_dir, "Library", "bin")
-        if os.path.exists(expected_dir):
-            os.add_dll_directory(expected_dir)
+        # Expect to find file per PEP-0405
+        expected_file = os.path.join(venv_dir, "pyvenv.cfg")
+        dll_dir = os.path.join(venv_dir, "Library", "bin")
+        if os.path.isdir(dll_dir) and os.path.isfile(expected_file):
+            os.add_dll_directory(dll_dir)
```

Comment:
If I'm understanding PEP-0405 correctly (and according to some [StackOveflow discussion](https://stackoverflow.com/questions/1871549/how-to-determine-if-python-is-running-inside-a-virtualenv)) another approach could be (per Python docs) use of [base_exec_prefix](https://docs.python.org/3/library/sys.html#sys.base_exec_prefix) and [exec_prefix](https://docs.python.org/3/library/sys.html#sys.exec_prefix), for example:
```
import sys
...
if sys.base_exec_prefix != sys.exec_prefix:
        dll_dir = os.path.join(sys.base_exec_prefix, "Library", "bin")
```

Reply:
Great idea. That way we do not depend on the environment variable, closing possible security hole.